src/main.rs

```diff
@@ -85,9 +105,14 @@ async fn main() -> Result<()> {
 
     let config = load_config()?;
 
-    global::ensure_global_config()?;
+    let global_config = global::ensure_global_config()?;
+
+    // init
+    let meter_provider = telemetry::init_telemetry(global_config.telemetry.otlp_endpoint);
+
+    let command_name = cli.command.name();
```

Comment:
the flag saying if telemetry is on should be used here to avoid bootstrapping the whole otlp infra if we're not going to push any data.